src/utils/sanitizer.ts

```
@@ -1,191 +1,194 @@
-        };
+    const trimmed = imageUrl.trim();
+
+    // Detect data URLs
+    if (/^data:/i.test(trimmed)) {
+        const mimeMatch = trimmed.match(/^data:([^;,]+)[;,]/i);
+        const mime = mimeMatch ? mimeMatch[1].toLowerCase() : "";
+        if (mime.startsWith("image/")) {
+            // Ensure the data URL is base64‑encoded
+            // The pattern checks for a ';base64,' marker after the MIME type
+            // Validate base64 data URL: ensure it contains only valid base64 characters after the comma
+            const base64Match = trimmed.match(/^data:image\/[^;]+;base64,([^]*)$/i);
```

Biome (2.1.2)

[error] The regular expression includes this negated empty character class.

Negated empty character classes match anything.
If you want to match against [, escape it \[.
Otherwise, remove the character class or fill it.

(lint/correctness/noEmptyCharacterClassInRegex)